check-peps.py

```diff
@@ -422,19 +422,41 @@ def _validate_post_history(line_num: int, body: str) -> MessageIterator:
                 yield from _date(offset, post_date, "Post-History")
                 yield from _thread(offset, post_url, "Post-History")
             else:
-                yield offset, f"post line must be a date or both start with “`” and end with “>`__”"
+                yield offset, "post line must be a date or both start with “`” and end with “>`__”"
 
 
 def _validate_resolution(line_num: int, line: str) -> MessageIterator:
-    """'Resolution' must be a direct thread/message URL"""
-
-    yield from _thread(line_num, line, "Resolution", allow_message=True)
+    """'Resolution' must be a direct thread/message URL or a link with a date."""
+
+    prefix, postfix = (line.startswith("`"), line.endswith(">`__"))
+    if not prefix and not postfix:
+        yield from _thread(line_num, line, "Resolution", allow_message=True)
+    elif prefix and postfix:
+        post_date, post_url = line[1:-4].split(" <")
+        yield from _date(line_num, post_date, "Resolution")
+        yield from _thread(line_num, post_url, "Resolution", allow_message=True)
+    else:
+        yield line_num, "Resolution line must be a link or both start with “`” and end with “>`__”"
 
 
 ########################
 #  Validation Helpers  #
 ########################
 
+def _validate_link_or_thread(text: str, header: str, offset: int, *, allow_message: bool = False) -> MessageIterator:
```

Comment:
Is this function used anywhere?

Reply:
Oops, forgot to remove it. (I tried to create shared logic for the Post-History and Resolution fields but ultimately decided they were too different and it was better to keep two separate functions.)